var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -56,9 +58,12 @@ class Caliper(CMakePackage):
     # gotcha doesn't work on Mac
     variant('gotcha', default=sys.platform != 'darwin',
             description='Enable GOTCHA support')
+    variant('sampler', default=sys.platform == 'linux2',
```

Comment:
`linux2` isn't reliable:
https://stackoverflow.com/questions/10415942/python-sys-platform-linux2-but-not-linux3

It sounds like `sys.platform.startswith('linux')` is more reliable.